### ISSUE TITLE
document-start rule with present: false shouldn't complain about the 2nd+ separators

### ISSUE THREAD

**Issue by adamchainz** [adrienverge/yamllint#115]
If you want your multi-document yaml files to look like:

```yaml
1
---
2
---
3
```

You might think it reasonable to use `document-start: {present: false}`, but unfortunately this throws an error for each of the `---` - even though there's no other way to express the file.

**Comment by adrienverge** [comment]
I agree the rule could be improved.

But today `document-start: {present: false}` can also be used to prevent documents starts *in order to prevent multi-documents* in YAML files.

What about something like this? Do you see any corner cases where it would not work / lead to inconsistencies?
```yaml
document-start:
  present: false
  allow-multiple-documents: true
```

**Comment by pixelastic** [comment]
I was having the same issue and your `document-start` proposal makes sense to me. At least it would allow me to have a linting configuration synchronized with what [prettier](https://github.com/prettier/prettier) outputs for yaml files.

Either this or a top-level `multiple-documents: true/false` new rule to prevent multi-document yaml.

**Comment by flmmartins** [comment]
Having the same issue on yamlling 1.16.0 in multi document even with present: false

**Comment by gdubicki** [comment]
+1 for adding a rule `allow-multiple-documents` that I could set to false. I am considering implementing this myself ..unless anyone here is already working on it?

**Comment by james-callahan** [comment]
> a top-level `multiple-documents: true/false` new rule to prevent multi-document yaml.

I think this would make more sense.
